primitives/weights/src/weight_v2.rs

```diff
@@ -325,12 +410,18 @@ impl Bounded for Weight {
 
 impl AddAssign for Weight {
 	fn add_assign(&mut self, other: Self) {
-		*self = Self { ref_time: self.ref_time + other.ref_time };
+		*self = Self {
+			ref_time: self.ref_time + other.ref_time,
+			proof_size: self.proof_size + other.proof_size,
+		};
```

Comment:
Is Rust not default implementing these?!

```suggestion
		*self = *self + other
```